frontend.py

```diff
@@ -0,0 +1,127 @@
+#! /usr/bin/env python3
+
+import collections
+from email.message import EmailMessage
+import http.server
+import json
+import socketserver
+import uuid
+
+import mailqueue
+
+
+mq_manager = mailqueue.Manager()
+
+
+class ThreadedHTTPServer(socketserver.ThreadingMixIn, http.server.HTTPServer):
+    """This server spawns a new thread for each incoming request"""
+    pass
+
+
+class Handler(http.server.BaseHTTPRequestHandler):
+    def do_POST(self):
+        if self.path == '/send':
+            self._handle_post_with(SendHandler)
+        elif self.path == '/status':
+            self._handle_post_with(StatusHandler)
+        else:
+            self.send_error(404, 'Not found')
+
+    def _handle_post_with(self, handler_class):
```

Comment:
It's a pretty smart way to handle requests. It's pretty clean and there is no code duplication. But I'd still rather see ``response_data = SendHandler.run(body)`` than ``response_data = handler_class(mq_manager).run(body)``. It can result in more code and even in code duplication but I'd have to think less :)

Reply:
Hmm, I'd be happy to improve readability (although the way it's done now doesn't strike me as poorly readable), and to do that I'd like to understand what aspect of the code seems stumbling to you. Is it passing the class object around? Or a chained call to the class constructor and then `.run()`?

Will something like this look better:

```
    def do_POST(self):
        if self.path == '/send':
            handler = SendHandler(mq_manager)
        elif self.path == '/status':
            handler = StatusHandler(mq_manager)
        else:
            self.send_error(404, 'Not found')

       self._handle_post_with(handler)
```
?

Reply:
Passing the class object. The chained call is fine. Yep, I like suggested variant better.

Reply:
Done!